nemo/collections/multimodal/modules/stable_diffusion/attention.py

```diff
@@ -106,8 +107,12 @@ def __init__(self, dim_in, dim_out):
         self.proj = LinearWrapper(dim_in, dim_out * 2)
 
     def forward(self, x):
-        x, gate = self.proj(x).chunk(2, dim=-1)
-        return x * F.gelu(gate)
+        x = self.proj(x)
+        if x.is_cuda and x.dtype == torch.float16 and x.shape[-1] // 2 in [1280, 2560, 5120] and x.is_contiguous():
```

Comment:
Is there a way we can hide the logic here in `fast_geglu` somewhere? and if the conditions are not met, we fallback to `F.gelu()`?

Reply:
Done